huaweicloud/services/dms/data_source_huaweicloud_dms_kafka_smart_connect_tasks.go

```diff
@@ -0,0 +1,207 @@
+			},
+		},
+	}
+}
+
+func tasksSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the ID of the smart connect task.`,
+			},
+			"task_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the name of the smart connect task.`,
+			},
+			"destination_type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the destination type of the smart connect task.`,
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the creation time of the smart connect task.`,
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the status of the smart connect task.`,
+			},
+			"topics": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the topic names separated by commas or the topic regular expression of the smart connect task.`,
+			},
+		},
+	}
+	return &sc
+}
+
+func resourceDmsKafkaSmartConnectTasksRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getKafkaSmartConnectTasks: query DMS kafka smart connect tasks
+	var (
+		getKafkaSmartConnectTasksHttpUrl = "v2/{project_id}/connectors/{connector_id}/sink-tasks"
+		getKafkaSmartConnectTasksProduct = "dms"
+	)
+	getKafkaSmartConnectTasksClient, err := cfg.NewServiceClient(getKafkaSmartConnectTasksProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DMS Client: %s", err)
+	}
+
+	connectorID := d.Get("connector_id").(string)
+	getKafkaSmartConnectTasksPath := getKafkaSmartConnectTasksClient.Endpoint + getKafkaSmartConnectTasksHttpUrl
+	getKafkaSmartConnectTasksPath = strings.ReplaceAll(getKafkaSmartConnectTasksPath, "{project_id}",
+		getKafkaSmartConnectTasksClient.ProjectID)
+	getKafkaSmartConnectTasksPath = strings.ReplaceAll(getKafkaSmartConnectTasksPath, "{connector_id}", connectorID)
+
+	getKafkaSmartConnectTasksOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+	getKafkaSmartConnectTasksResp, err := getKafkaSmartConnectTasksClient.Request("GET", getKafkaSmartConnectTasksPath,
+		&getKafkaSmartConnectTasksOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving DMS kafka smart connect tasks")
+	}
+
+	getKafkaSmartConnectTasksRespBody, respBodyerr := utils.FlattenResponse(getKafkaSmartConnectTasksResp)
+	if respBodyerr != nil {
+		return diag.FromErr(respBodyerr)
+	}
+
+	dataSourceId, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(dataSourceId)
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("connector_id", connectorID),
```

Comment:
it's unnecessary to set the required param in a data source.

Reply:
deleted connector_id